### DISCUSSION TITLE
Oqtane.Modules.Controls.TabPanel: call function on click over Tab?

### DISCUSSION THREAD

**Discussion by artmedia** [oqtane/oqtane.framework#1083] [Q&A]
I just started using Oqtane's component TabStrip on my component and I was trying to execute a function on click event over a Tab.
I get this error: Object of type 'Oqtane.Modules.Controls.TabPanel' does not have a property matching the name 'onclick'
Is it possible to do that?

< TabStrip >
    < TabPanel Name="Utenti" @**onclick**="async () => await initializeCmpUsers()" >
        < CmpUsers @ref=" myCmpUsers" ></ CmpUsers >
    < /TabPanel >
    < TabPanel Name="Abbonamenti" @**onclick**="async () => await initializeCmpAbbonamenti()" >
        < CmpAbbonamenti @ref=" myCmpAbbonamenti" >< /CmpAbbonamenti >
    < /TabPanel >
< /TabStrip >

thanks in advance

**Comment by sbwalker** [comment]
The TabPanel component does not currently support a click event. The TabStrip and TabPanel simply wrap the Bootstrap nav feature and rely on its client-side functionality.

**Comment by hishamco** [comment]
May be we could introduce a `onTabChange` event for some scenarios like data loading for a particular tab

**Reply by W6HBR** [reply]
I'm dealing with this exact scenario right now.  I would like to trigger a data load/query upon activating a particular tab.  Without an event here, all data for my tabs has to be queried upon page load which slows down the initial load/responsiveness.

onTabChange would be most welcome!
